packages/react-native-reanimated/Common/cpp/Fabric/ReanimatedCommitHook.cpp

```diff
@@ -38,23 +40,21 @@ RootShadowNode::Unshared ReanimatedCommitHook::shadowTreeWillCommit(
   // ShadowTree not commited by Reanimated, apply updates from PropsRegistry
 
   auto rootNode = newRootShadowNode->ShadowNode::clone(ShadowNodeFragment{});
+  std::unordered_map<
+      const ShadowNodeFamily *,
+      std::vector<std::shared_ptr<RawProps>>>
```

Comment:
Can we use `std::unique_ptr` instead of `std::shared_ptr` here? Also, it's not recommended to allocate `RawProps` on heap and hold it for too long, can we avoid using pointers here at all?

Reply:
In the `ReanimatedCommitHook` we have props stored as `folly::dynamic` and in `performOperations` we have them as `jsi::Value`. We have to turn them into `RawProps` at some point before cloning `ShadowNodes`. We could use some abstraction over `jsi::Value` and `folly::dynamic` but this is exactly what `RawProps` do. They need to be allocated on the heap, because we want to process the entire batch efficiently. Also, we only hold on to them for the duration of the `clone` function, we never persist them across multiple invocations.

Reply:
Nevermind

Reply:
Never-nevermind